### DISCUSSION TITLE
how to change devices?

### DISCUSSION THREAD

**Discussion by zepadovani** [smrg-lm/sc3#11] [General]
Hi! Congratulations on the project: it looks very promising!

I'm struggling here to change the devices... the default ones have different sample rates and thus I cannot boot (have tried to change my OS default also)

I have seen in server.py that currently it is not possible to use attributes like s.option.inDevice and s.option.outDevice... are there any workarounds?

thanks!

**Comment by smrg-lm** [comment]
Hi, I'll take a look, the attribute of ServerOptions is `hw_device_name` but doesn't separate in input and output devices and doesn't quotes the cmd string I see. You may try `s.options.hw_device_name = ' "My Device" '` (double quotes within single quotes) or `s.options.hw_device_name = ' "My InDevice" "My OutDevice '`,  but I have to solve it.

**Comment by smrg-lm** [comment]
Just pushed a simple fix, note that it is not possible to "programmatically obtain a list of available audio devices" because it would require an external dependency and so far this lib has none.

```
s.options.hw_device_name = 'IO Device Name'
```
or
```
s.options.hw_device_name = ('In Device Name', 'Out Device Name')
```

**Comment by zepadovani** [comment]
Wonderful! thanks! I will take a look soon!
